static/js/settings_account.js

```diff
@@ -122,6 +122,22 @@ exports.initialize_custom_date_type_fields = function (element_id) {
     $(element_id).find(".custom_user_field .datepicker").flatpickr({
         altInput: true,
         altFormat: "F j, Y"});
+
+    if (!$(element_id).find(".custom_user_field .datepicker").val()) {
+        $(element_id).find(".remove_date").hide();
+    }
+
+    $(element_id).find(".custom_user_field .datepicker").mouseenter(function () {
+        if (!$(element_id).find(".custom_user_field .datepicker").val()) {
+            $(element_id).find(".remove_date").hide();
+        } else {
+            $(element_id).find(".remove_date").show();
+        }
+    });
+
+    $(element_id).find(".remove_date").click(function () {
+        $(element_id).find(".custom_user_field .datepicker").flatpickr();
```

Comment:
I myself didn't understood but without its usage, the remove_date is not working I think maybe its due to some issue in function calling

Reply:
I tested the version, it is working at my end. Can you post gif of bug?

Reply:
Yes Sure
![Peek 2019-04-07 18-44](https://user-images.githubusercontent.com/18120004/55684110-4a063300-5965-11e9-9d81-15a45602930e.gif)
In this, I am clicking on X button but it isn't working.

Reply:
Okay, it seems like it is working on master branch without event handler on `remove_date`. But that is really buggy, sometime it works and sometime not. So I think adding event handler is safe option.

Reply:
If it is nondeterministic, I'd look closely at the existing code -- it's possible we have something like "if you open users before account settings" being the variable for whether it works (e.g. because the "account settings" code has a too-broad selector).

Reply:
@vipulchhabra99 I am assuming that you are investigating this issue.

Reply:
@YJDave i am also unable to figure out why handler is not working and what are method left to change the color of field